src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -208,65 +180,131 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._pool.close()
             self._pool.join()
 
-    def _timeout_process_wrapper(self, run_start_time: datetime, task_queue: Queue, timeout_time, result_list):
-        healthy_ensured_process = HealthyEnsuredProcess(self._executor_creation_func, self.context)
-        healthy_ensured_process.start_new(task_queue)
-
-        if not healthy_ensured_process.process.is_alive():
-            return
-
+    def _process_task(
+        self,
+        process_info,
+        task_queue,
+        timeout_time,
+        result_list
+    ):
         while True:
+            index, process_id, process_name, input_queue, output_queue = (
+                process_info.index,
+                process_info.process_id,
+                process_info.process_name,
+                process_info.input_queue,
+                process_info.output_queue
+            )
+
             try:
                 args = task_queue.get(timeout=1)
             except queue.Empty:
-                healthy_ensured_process.end()
+                self._processes_manager.end_process(index)
                 return
 
-            healthy_ensured_process.put(args)
+            input_queue.put(args)
             inputs, line_number, run_id = args[:3]
-            self._processing_idx[line_number] = healthy_ensured_process.format_current_process(line_number)
 
+            self._processing_idx[line_number] = format_current_process(process_name, process_id, line_number)
             start_time = datetime.utcnow()
             completed = False
+            crashed = False
 
             while datetime.utcnow().timestamp() - start_time.timestamp() <= timeout_time:
                 try:
+                    # Monitor process aliveness.
+                    if not psutil.pid_exists(process_id):
+                        crashed = True
+                        break
+
                     # Responsible for checking the output queue messages and
                     # processing them within a specified timeout period.
-                    message = healthy_ensured_process.get()
-                    if isinstance(message, LineResult):
-                        completed = True
-                        message = self._process_multimedia(message)
-                        result_list.append(message)
+                    message = output_queue.get(timeout=1)
+                    completed = self._process_output_message(message, result_list)
```

Comment:
It is a little confusing the function "process_output_message" returns a boolean value. Considering a more accurate function name or adjust the logic as follows:
```
self._process_output_message
if isinstance(message, LineResult)  # Add comments to explain why it is completed here
    break
```
```

Reply:
completed = self._handle_output_queue_messages(output_queue, result_list)
    if completed:
        break

rename the function